### ISSUE TITLE
EfficientDet keras profiling

### ISSUE THREAD

**Issue by rkaskman** [google/automl#1066]
I tried profiling EfficientDet via tf.keras.callbacks.TensorBoard and judging by the results, it fails to capture the real execution times of different steps. 

![image](https://user-images.githubusercontent.com/4347953/127972271-1266a8f9-ea8c-44d1-b652-26bc9ec8d558.png)

 According to the logs, CUPTI is found and profiler events are getting recorded. 

`2021-08-02 18:48:30.803744: I tensorflow/core/profiler/lib/profiler_session.cc:164] Profiler session started.
2021-08-02 18:48:30.803986: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1391] Profiler found 2 GPUs
2021-08-02 18:48:30.809586: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cupti64_101.dll
2021-08-02 18:48:30.933819: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1513] CUPTI activity buffer flushed
2021-08-02 18:50:01.937082: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cudnn64_7.dll
2021-08-02 18:50:09.493744: W tensorflow/stream_executor/gpu/redzone_allocator.cc:314] Internal: Invoking GPU asm compilation is supported on Cuda non-Windows platforms only
Relying on driver to perform ptx compilation. 
Modify $PATH to customize ptxas location.
This message will be only logged once.
2021-08-02 18:50:09.562346: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cublas64_10.dll
19/60 [========>.....................] - ETA: 42s - det_loss: 463.2457 - cls_loss: 462.4496 - box_loss: 0.0159 - reg_l2_loss: 0.0977 - loss: 463.3435 - learning_rate: 0.0083 - gradient_norm: 8.64092021-08-02 18:50:33.068281: I tensorflow/core/profiler/lib/profiler_session.cc:164] Profiler session started.
39/60 [==================>...........] - ETA: 23s - det_loss: 226.6503 - cls_loss: 225.9500 - box_loss: 0.0140 - reg_l2_loss: 0.0978 - loss: 226.7480 - learning_rate: 0.0086 - gradient_norm: 5.25052021-08-02 18:51:00.374828: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1513] CUPTI activity buffer flushed
2021-08-02 18:51:04.055234: I tensorflow/core/profiler/internal/gpu/device_tracer.cc:223]  GpuTracer has collected 369743 callback api events and 369756 activity events. 
2021-08-02 18:51:20.080024: I tensorflow/core/profiler/rpc/client/save_profile.cc:176] Creating directory: C:/Projects/ed_profiler_debug\train\plugins\profile\2021_08_02_16_51_12
2021-08-02 18:51:24.068445: I tensorflow/core/profiler/rpc/client/save_profile.cc:182] Dumped gzipped tool data for trace.json.gz to C:/Projects/ed_profiler_debug\train\plugins\profile\2021_08_02_16_51_12\workstation.trace.json.gz
2021-08-02 18:51:33.536592: I tensorflow/core/profiler/rpc/client/save_profile.cc:176] Creating directory: C:/Projects/ed_profiler_debug\train\plugins\profile\2021_08_02_16_51_12
2021-08-02 18:51:33.557332: I tensorflow/core/profiler/rpc/client/save_profile.cc:182] Dumped gzipped tool data for memory_profile.json.gz to C:/Projects/ed_profiler_debug\train\plugins\profile\2021_08_02_16_51_12\workstation.memory_profile.json.gz
2021-08-02 18:51:35.268901: I tensorflow/python/profiler/internal/profiler_wrapper.cc:111] Creating directory: C:/Projects/ed_profiler_debug\train\plugins\profile\2021_08_02_16_51_12
Dumped tool data for xplane.pb to C:/Projects/ed_profiler_debug\train\plugins\profile\2021_08_02_16_51_12\workstation.xplane.pb
Dumped tool data for overview_page.pb to C:/Projects/ed_profiler_debug\train\plugins\profile\2021_08_02_16_51_12\workstation.overview_page.pb
Dumped tool data for input_pipeline.pb to C:/Projects/ed_profiler_debug\train\plugins\profile\2021_08_02_16_51_12\workstation.input_pipeline.pb
Dumped tool data for tensorflow_stats.pb to C:/Projects/ed_profiler_debug\train\plugins\profile\2021_08_02_16_51_12\workstation.tensorflow_stats.pb
Dumped tool data for kernel_stats.pb to C:/Projects/ed_profiler_debug\train\plugins\profile\2021_08_02_16_51_12\workstation.kernel_stats.pb`

**Comment by danielTobon43** [comment]
@rkaskman how did you achieve the profiler tool. I have set the `--profile` flag and a `profile_batch=2` and I am not getting anything in tensorboard.
